cg/store/models.py

```diff
@@ -628,91 +692,95 @@ def to_dict(self):
-    customer_id = Column(ForeignKey("customer.id", ondelete="CASCADE"), nullable=False)
-    customer = orm.relationship(Customer, foreign_keys=[customer_id])
-    delivered_at = Column(types.DateTime)
-    deliveries = orm.relationship(Delivery, backref="pool")
-    id = Column(types.Integer, primary_key=True)
-    invoice_id = Column(ForeignKey("invoice.id"))
-    name = Column(types.String(32), nullable=False)
-    no_invoice = Column(types.Boolean, default=False)
-    order = Column(types.String(64), nullable=False)
-    ordered_at = Column(types.DateTime, nullable=False)
-    received_at = Column(types.DateTime)
-    ticket = Column(types.String(32))
-
-    invoice = orm.relationship("Invoice", back_populates="pools")
+    comment: Mapped[Optional[text]]
+    created_at: Mapped[Optional[datetime]] = mapped_column(default=datetime.now)
+    customer_id: Mapped[int] = mapped_column(ForeignKey("customer.id", ondelete="CASCADE"))
+    customer: Mapped["Customer"] = orm.relationship(foreign_keys=[customer_id])
+    delivered_at: Mapped[Optional[datetime]]
+    deliveries: Mapped[list["Delivery"]] = orm.relationship(backref="pool")
+    id: Mapped[intpk]
+    invoice_id: Mapped[Optional[int]] = mapped_column(ForeignKey("invoice.id"))
+    name: Mapped[str32]
+    no_invoice: Mapped[Optional[bool]] = mapped_column(default=False)
+    order: Mapped[str64]
+    ordered_at: Mapped[datetime]
+    received_at: Mapped[Optional[datetime]]
+    ticket: Mapped[Optional[str32]]
+
+    invoice: Mapped[Optional["Invoice"]] = orm.relationship(back_populates="pools")
 
     def to_dict(self):
         return to_dict(model_instance=self)
 
 
-class Sample(Model, PriorityMixin):
+class Sample(Base, PriorityMixin):
     __tablename__ = "sample"
-    age_at_sampling = Column(types.FLOAT)
-    application_version_id = Column(ForeignKey("application_version.id"), nullable=False)
-    application_version = orm.relationship(
-        ApplicationVersion, foreign_keys=[application_version_id]
+    age_at_sampling: Mapped[Optional[float]]
+    application_version_id: Mapped[int] = mapped_column(ForeignKey("application_version.id"))
+    application_version: Mapped["ApplicationVersion"] = orm.relationship(
+        foreign_keys=[application_version_id]
+    )
+    capture_kit: Mapped[Optional[str64]]
+    comment: Mapped[Optional[text]]
+    control: Mapped[Optional[ControlOptions]]
+    created_at: Mapped[Optional[datetime]] = mapped_column(default=datetime.now)
+    customer_id: Mapped[int] = mapped_column(ForeignKey("customer.id", ondelete="CASCADE"))
+    customer: Mapped["Customer"] = orm.relationship(foreign_keys=[customer_id])
+    delivered_at: Mapped[Optional[datetime]]
+    deliveries: Mapped[list["Delivery"]] = orm.relationship(backref="sample")
+    downsampled_to: Mapped[Optional[bigint]]
+    from_sample: Mapped[Optional[str128]]
+    id: Mapped[intpk]
+    internal_id: Mapped[str32] = mapped_column(unique=True)
```

Comment:
If we want to extend this we could for example create the type int `uniq_str` an just have this in several classes:
`internal_id: Mapped[uniq_str]`

Reply:
Do we need multiple that specifcy char length, or are we certain unique strings will never be longer than a certain length or do we leave them uncapped?

Reply:
It needs to map to a specific String sub-type for it to work. Haven't looked into how many different one we use for this purpose. But glad you like it. Will look into it a bit

Reply:
Most of them where String(32). Created and implemented the type hint `StrUniq`